bot/exts/help_channels.py

```diff
@@ -213,18 +213,20 @@ async def close_command(self, ctx: commands.Context) -> None:
         and reset the send permissions cooldown for the user who started the session.
         """
         log.trace("close command invoked; checking if the channel is in-use.")
-        if ctx.channel.category == self.in_use_category:
-            if await self.dormant_check(ctx):
-                await self.remove_cooldown_role(ctx.author)
+        if ctx.channel.category != self.in_use_category:
+            log.debug(f"{ctx.author} invoked command 'dormant' outside an in-use help channel")
+            return
+
+        if not await self.dormant_check(ctx):
+            return
```

Comment:
Is there a reason why it's wrong in this particular case?

Reply:
It is not really wrong. I mainly found it unnecessary. The suggestion at the end is mainly because it'd make the code shorter:

```py
if not await self.dormant_check(ctx):
    return

await self.move_to_dormant(ctx.channel, "command")
self.scheduler.cancel(ctx.channel.id)
```

versus

```py
if await self.dormant_check(ctx):
    await self.move_to_dormant(ctx.channel, "command")
    self.scheduler.cancel(ctx.channel.id)
```

It also makes the control flow slightly easier to follow.

Reply:
I think early exiting makes it easier to tell which lines are doing what the function is generally expected to do versus which lines are handling a special or unexpected case, though I'll defer to your judgement.